### PR TITLE
Add Hugo based web site.

### PR DESCRIPTION
**Pull request by sophokles73**
I have added folder `site/` which contains the source code for a Californium website to be hosted at https://www.eclipse.org/californium`. The site is based on the [Hugo](https://gohugo.io) system which is use in many other Eclipse projects as well.
It requires the hugo binary to be installed for generating the static HTML pages from the source code.
See README.md for details how to build the website locally. I have tried to include all/most of the content from the existing website.

### CODE REVIEW

**Comment by boaks** [comment]
I'm right, that the README.md is a general one, which includes plenty of possible ways to install?
Trying just apt install hugo results in a too old version. 
Any recommendation, which of the approaches for installing hugo works most easily?

**Comment by sophokles73** [comment]
@boaks simply download the latest version from https://github.com/gohugoio/hugo/releases

If you ar on Ubuntu you might also want to try using the Snap
```
snap install hugo
```

**Comment by boaks** [comment]
Thanks. I managed to install hugo and run the page with "hugo server -D".
Great work!
You removed the "copper" link. Is that by intention?

**Comment by sophokles73** [comment]
> You removed the "copper" link. Is that by intention?

Yes, because you marked it as *hibernated*, I was wondering if we still want it to be on the page. It is very little work to add it. I just wanted to make sure that we think it has value, given that it basically doesn't work anymore in any modern browser, or am I mistaken?

File: site/content/community/channels/mailing-list.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+weight = 20
+
+[asset]
+  icon = "fas fa-envelope-square"
+  url = "https://accounts.eclipse.org/mailing-list/cf-dev"
++++
+
+An excellent way to get the latest news and announcements around Californium is to join the mailing list.
+
+This is the place where we announce new releases and discuss architectural issues of greater impact.
```
Comment (sbernard31) [inline site/content/community/channels/mailing-list.md]: I'm not sure the mailing is so much used ?

Comment (sophokles73) [reply site/content/community/channels/mailing-list.md]: Would you like to change the wording only or do you want to remove the mailing list altogether?

File: site/content/_index/hero.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
++++
+fragment = "hero"
+#disabled = true
+date = "2019-10-29"
+weight = 50
+background = "light" # can influence the text color
+particles = false
+
+title = "Hello"
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply site/content/_index/hero.md]: If we want to keep the backdrop image then the title is ignored (and can be removed). If you would rather remove the backdrop then the title is used if I am not mistaken

File: site/content/community/channels/index.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
++++
+fragment = "items"
+weight = 150
+title = "Get in Touch"
+subtitle = "Wanna tell us what you think about Eclipse Californium&trade;? What we can improve? Which feature is missing? Or just discuss architectural issues?"
+background = "dark"
++++
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply site/content/community/channels/index.md]: We can also include it in the main page. No problem. For the time being, there is not much content anyway so having everything on a single page might indeed make more sense.

File: site/content/_index/hero.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#subtitle = "Eclipse Californium&trade;"
+
+[header]
+  image = "java.jpg"
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply site/content/_index/hero.md]: @boaks what do you think? With or without the backdrop?

[the remark this answers is not in the document]
Comment (boaks) [reply site/content/_index/hero.md]: no preference.

**Comment by sophokles73** [comment]
@sbernard31 I have moved the community section to the landing page. We could simply drop the first paragraph of the mailing description, would that work for you?

I have also removed the backdrop. Which background color do you like better, *light* or *secondardy*?

**Comment by boaks** [comment]
The top field grey with only the Californium logo makes me feel, that there is something missing.
Is it possible to move the logo on field down and drop that grey?

File: site/content/community/channels/mailing-list.md (hunk 1 above)
Comment (sbernard31) [reply site/content/community/channels/mailing-list.md]: My idea : if nobody use it, we can remove it.

But If you think, we must keep it. It's OK for me too.

Comment (boaks) [reply site/content/community/channels/mailing-list.md]: Yes, it's really rare used.
I would keep it, for those, who don't have a github account (even if they area also rare :-) .
Explaining, that github issues (or eclipse tickets) are usually better for focus on discussing a topic.

File: site/content/community/channels/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sbernard31) [reply site/content/community/channels/index.md]: :+1:

**Comment by boaks** [comment]
For me the boxes (with Californium, Actinium, Scandium) are optical more attractive, than the logo on grey on to much space :-).
But anyway, I think, go for it, and improve it ongoing.

**Comment by sophokles73** [comment]
> If someone has just the logo, I bet this should looks better. (more minimalist)

At least I don't have them. I'd say we start with the exiting *boxes* and once you, @sbernard31, have figured out how to extract the raw logos from the boxes, we'll use those instead ;-)

**Comment by sophokles73** [comment]
> The top field grey with only the Californium logo makes me feel, that there is something missing.

Are you talking about the header which includes the nav bar and the GitHub link at the right?

**Comment by sbernard31** [comment]
> once you, @sbernard31, have figured out how to extract the raw logos from the boxes, we'll use those instead ;-)

Don't challenge me or I will do it and we will finish with ugly logo instead of old school boxes. :grin:

**Comment by sophokles73** [comment]
@boaks 
I have replaced the logo in the header with the project name. Does that look better to you?

@sbernard31 
I have changed the wording for the mailing list. WDYT?

**Comment by boaks** [comment]
> I have replaced the logo in the header with the project name. Does that look better to you?

yes.

(Even if I had the large grey box with the large californium logo in mind. But, please go ahead and merge it.)

**Comment by boaks** [comment]
> once you, @sbernard31, have figured out how to extract the raw logos from the boxes, we'll use those instead ;-)

Sorry for the delay:
The logos are in the current webpage-repo, folder "dev/app/images/logos".
Cf, Ac, Sc, different sizes

Ac_512
Cf_512
Sc_512

**Comment by sophokles73** [comment]
I'll give it a shot then using the logos instead of the *dreaded* boxes ...

**Comment by sophokles73** [comment]
@sbernard31 @boaks 
I have pushed the updated site using the logos instead of the boxes. WDYT?

**Comment by boaks** [comment]
Great. I would prefer the project name on the upper left corner instead of the logo, as it was before the last change.

**Comment by sbernard31** [comment]
> WDYT?

It seems better like this to me.
(but no problem if you prefer keep the boxes)

**Comment by sophokles73** [comment]
@boaks looks like we're getting close now, I have changed the nav bar according to your preference ...

**Comment by boaks** [comment]
LGTM

**Pull request merged**